### DISCUSSION TITLE
Are the codes for hybrid search open source?

### DISCUSSION THREAD

**Discussion by Kimchuls** [milvus-io/milvus#28512] [Q&A and General discussion]
I cloned some code in Milvus repositories to find out the implementation of vector search, but I can only find the code for vector similarity search and those indexes. Is the code for hybrid search(vector search with attribute filtering) open source? If yes, where can I find it? Thank you.

**Comment by xiaofan-luan** [comment]
currently 2.3 support atrribute filtering.
see here https://milvus.io/docs/hybridsearch.md

**Reply by Kimchuls** [reply]
Actually, I am wondering whether I can see the code for implementing such a function in github repositories(like milvus-io/milvus or milvus-io/knowhere)? Or is the code for hybrid search is not open source?

**Reply by yiwangdr** [reply]
Yes it's open sourced.
https://github.com/milvus-io/milvus/tree/82915a9630ab0ff40d7891b97c367ede5726ff7c/internal/core/src
I would look into the index/ and query/ (e.g. trace how [expression](https://github.com/milvus-io/milvus/blob/82915a9630ab0ff40d7891b97c367ede5726ff7c/internal/core/src/query/Expr.h#L59) is used)

**Reply by xiaofan-luan** [reply]
you are talking about a long stroage.

it start from parser in golang, filtering code in segcore and vector index code in knowhere

**Reply by Kimchuls** [reply]
Thank you very much!
